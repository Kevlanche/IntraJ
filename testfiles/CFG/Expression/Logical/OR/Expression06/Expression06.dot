digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "MethodDecl[12]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Modifiers[14]"; "PrimitiveTypeAccess[16]"; "List[17]"; "List[30]"; "Opt[31]"; "Exit[32]"}
{ rank = same; "List[15]"; "ParameterDeclaration[18]"; "ParameterDeclaration[22]"; "ParameterDeclaration[26]"; "Block[32]"}
{ rank = same; "Modifiers[19]"; "PrimitiveTypeAccess[21]"; "Modifiers[23]"; "PrimitiveTypeAccess[25]"; "Modifiers[27]"; "PrimitiveTypeAccess[29]"; "List[33]"}
{ rank = same; "List[20]"; "List[24]"; "List[28]"; "ForStmt[34]"}
{ rank = same; "List[35]"; "Opt[36]"; "List[42]"; "Block[43]"}
{ rank = same; "AssignSimpleExpr[37]"; "List[44]"}
{ rank = same; "VarAccess[38]"; "OrLogicalExpr[39]"}
{ rank = same; "VarAccess[40]"; "VarAccess[41]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < Expression06 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[12]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[14]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[16]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[17]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[18]" [label="ParameterDeclaration\n < a >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[19]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[20]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[21]" [label="PrimitiveTypeAccess\n < boolean >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[22]" [label="ParameterDeclaration\n < b >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[23]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[24]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[25]" [label="PrimitiveTypeAccess\n < boolean >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[26]" [label="ParameterDeclaration\n < c >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[27]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[28]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[29]" [label="PrimitiveTypeAccess\n < boolean >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[31]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[32]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[33]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ForStmt[34]" [label="ForStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[35]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[36]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"AssignSimpleExpr[37]" [label="AssignSimpleExpr\n < a >", fillcolor=white   style=filled  ]
"VarAccess[38]" [label="VarAccess\n < a >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"OrLogicalExpr[39]" [label="OrLogicalExpr\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[40]" [label="VarAccess\n < b >", fillcolor=white   style=filled  ]
"VarAccess[41]" [label="VarAccess\n < c >", fillcolor=white   style=filled  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[43]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[44]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[32]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"Modifiers[14]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Modifiers[14]" [style=dashed, color=gray]
"MethodDecl[12]" -> "PrimitiveTypeAccess[16]" [style=dashed, color=gray]
"Modifiers[19]" -> "List[20]" [style=dashed, color=gray]
"ParameterDeclaration[18]" -> "Modifiers[19]" [style=dashed, color=gray]
"ParameterDeclaration[18]" -> "PrimitiveTypeAccess[21]" [style=dashed, color=gray]
"List[17]" -> "ParameterDeclaration[18]" [style=dashed, color=gray]
"Modifiers[23]" -> "List[24]" [style=dashed, color=gray]
"ParameterDeclaration[22]" -> "Modifiers[23]" [style=dashed, color=gray]
"ParameterDeclaration[22]" -> "PrimitiveTypeAccess[25]" [style=dashed, color=gray]
"List[17]" -> "ParameterDeclaration[22]" [style=dashed, color=gray]
"Modifiers[27]" -> "List[28]" [style=dashed, color=gray]
"ParameterDeclaration[26]" -> "Modifiers[27]" [style=dashed, color=gray]
"ParameterDeclaration[26]" -> "PrimitiveTypeAccess[29]" [style=dashed, color=gray]
"List[17]" -> "ParameterDeclaration[26]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[17]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[30]" [style=dashed, color=gray]
"ForStmt[34]" -> "List[35]" [style=dashed, color=gray]
"AssignSimpleExpr[37]" -> "VarAccess[38]" [style=dashed, color=gray]
"OrLogicalExpr[39]" -> "VarAccess[40]" [style=dashed, color=gray]
"OrLogicalExpr[39]" -> "VarAccess[41]" [style=dashed, color=gray]
"AssignSimpleExpr[37]" -> "OrLogicalExpr[39]" [style=dashed, color=gray]
"Opt[36]" -> "AssignSimpleExpr[37]" [style=dashed, color=gray]
"ForStmt[34]" -> "Opt[36]" [style=dashed, color=gray]
"ForStmt[34]" -> "List[42]" [style=dashed, color=gray]
"Block[43]" -> "List[44]" [style=dashed, color=gray]
"ForStmt[34]" -> "Block[43]" [style=dashed, color=gray]
"List[33]" -> "ForStmt[34]" [style=dashed, color=gray]
"Block[32]" -> "List[33]" [style=dashed, color=gray]
"Opt[31]" -> "Block[32]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Opt[31]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Exit[32]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[12]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[32]" -> "AssignSimpleExpr[37]" [color=red, constraint=false]
"AssignSimpleExpr[37]" -> "VarAccess[40]" [color=blue, constraint=false, xlabel="T "] 
"AssignSimpleExpr[37]" -> "Exit[32]" [color=blue, constraint=false, xlabel="F "] 
"AssignSimpleExpr[37]" -> "VarAccess[40]" [color=red, constraint=false]
"AssignSimpleExpr[37]" -> "VarAccess[41]" [color=red, constraint=false]
"VarAccess[41]" -> "AssignSimpleExpr[37]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[41]" -> "VarAccess[40]" [color=red, constraint=false]
"VarAccess[40]" -> "AssignSimpleExpr[37]" [color=blue, constraint=false, xlabel="T "] 
"VarAccess[40]" -> "VarAccess[41]" [color=blue, constraint=false, xlabel="F "] 
"VarAccess[40]" -> "Entry[13]" [color=red, constraint=false]
"VarAccess[40]" -> "AssignSimpleExpr[37]" [color=red, constraint=false]
"Entry[13]" -> "VarAccess[40]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"Entry[13]" -> "Modifiers[14]" -> "PrimitiveTypeAccess[16]" -> "List[17]" -> "List[30]" -> "Opt[31]" -> "Exit[32]" [style=invis]
"ParameterDeclaration[18]" -> "ParameterDeclaration[22]" -> "ParameterDeclaration[26]" [style=invis]
"Modifiers[19]" -> "PrimitiveTypeAccess[21]" [style=invis]
"Modifiers[23]" -> "PrimitiveTypeAccess[25]" [style=invis]
"Modifiers[27]" -> "PrimitiveTypeAccess[29]" [style=invis]
"List[35]" -> "Opt[36]" -> "List[42]" -> "Block[43]" [style=invis]
"VarAccess[38]" -> "OrLogicalExpr[39]" [style=invis]
"VarAccess[40]" -> "VarAccess[41]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
46", fillcolor=1]
Edges [label="Total Nr. Edges
57", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
5", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
6", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}