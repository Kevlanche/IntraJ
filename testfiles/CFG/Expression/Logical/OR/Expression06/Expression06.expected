NodeName: Entry[13] Successors: [VarAccess[40]]
NodeName: VarAccess[40] Successors: [AssignSimpleExpr[37], VarAccess[41]]
NodeName: AssignSimpleExpr[37] Successors: [VarAccess[40], Exit[32]]
NodeName: Exit[32] Successors: []
NodeName: VarAccess[41] Successors: [AssignSimpleExpr[37]]

NodeName: Exit[32] Predecessor: [AssignSimpleExpr[37]]
NodeName: AssignSimpleExpr[37] Predecessor: [VarAccess[40], VarAccess[41]]
NodeName: VarAccess[40] Predecessor: [Entry[13], AssignSimpleExpr[37]]
NodeName: Entry[13] Predecessor: []
NodeName: VarAccess[41] Predecessor: [VarAccess[40]]
